packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -665,14 +753,24 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     const linkMarks: GetMarkRange[] = [];
 
     if (from === to) {
-      const $pos = doc.resolve(from);
-      $pos.marks().forEach(() => {
-        const range = getMarkRange($pos, this.type);
+      const resolveFrom = Math.max(from - 1, 0);
 
-        if (range?.mark.attrs.auto === isAutoLink) {
-          linkMarks.push(range);
-        }
-      });
+      doc
+        .resolve(resolveFrom)
+        .marks()
+        .forEach((mark) => {
+          const $pos = doc.resolve(resolveFrom);
+
+          const range = getMarkRange($pos, mark.type);
```

Comment:
This looks like we duplicating work here? We've asserted `from === to` so were talking about an single position

I think we can get rid of the first doc resolve, and marks `forEach` here and just do

```ts
const $pos = doc.resolve(resolveFrom);
const range = getMarkRange($pos, mark.type);

if (range?.mark.attrs.auto === isAutoLink) {
  linkMarks.push(range);
}
```